Api/CalcWrapper.cs

```diff
@@ -0,0 +1,298 @@
+                        throw new unoidl.com.sun.star.uno.Exception(ex.Message, ex);
+                    }
+                }
+            }
+            return (unoidl.com.sun.star.sheet.XSpreadsheetDocument)xComponent;
+        }
+
+        private void SaveDocument(String Path)
+        {
+            try
+            {
+                Path = Path.Replace("\\", "/"); // Convert forward slashes to backslashes, converting it to the correct format storeToURL expects. 
+                Path = String.Concat("file:///", Path);
+                XStorable xStorable = (XStorable)mxDocument; // Typecast the currently open document to XStorable type.
+                xStorable.storeToURL(Path, new unoidl.com.sun.star.beans.PropertyValue[1]); //Creating an empty PropertyValue array saves the document in the default .ods format.
+            }
+            catch(unoidl.com.sun.star.uno.Exception ex)
+            {
+                throw new unoidl.com.sun.star.uno.Exception(ex.Message, ex);
+            }
+        }
+
+        private void AddNewSheet(String Name)
+        {
+            try
+            {
+                unoidl.com.sun.star.sheet.XSpreadsheets xSheets = mxDocument.getSheets();
+                short sheetIndex = Convert.ToInt16(xSheets.getElementNames().Length + 1); //Get the number of sheets already existing in the document, add one to get the new index. 
+                xSheets.insertNewByName(Name, sheetIndex);
+            }
+            catch(unoidl.com.sun.star.uno.Exception ex)
+            {
+                throw new unoidl.com.sun.star.uno.Exception(ex.Message, ex);
+            }
+        }
+
+        public void RemoveSheet(String Name)
+        {
+            try
+            {
+                unoidl.com.sun.star.sheet.XSpreadsheets xSheets = mxDocument.getSheets();
+                xSheets.removeByName(Name);
+            }
+            catch(unoidl.com.sun.star.uno.Exception ex)
+            {
+                throw new unoidl.com.sun.star.uno.Exception(ex.Message, ex);
+            }
+        }
+
+        private void SetActiveSheet(String sheetName)
+        {
+            try
+            {
+                unoidl.com.sun.star.sheet.XSpreadsheets xSheets = mxDocument.getSheets();
+                unoidl.com.sun.star.frame.XModel xModel = (unoidl.com.sun.star.frame.XModel)mxDocument;
+                unoidl.com.sun.star.frame.XController xController = xModel.getCurrentController();
+                unoidl.com.sun.star.sheet.XSpreadsheetView sheetView = (unoidl.com.sun.star.sheet.XSpreadsheetView)xController;
+                unoidl.com.sun.star.sheet.XSpreadsheet xSpreadsheet = (unoidl.com.sun.star.sheet.XSpreadsheet)xSheets.getByName(sheetName).Value;
+                sheetView.setActiveSheet(xSpreadsheet);
+            }
+           catch(unoidl.com.sun.star.uno.Exception ex)
+            {
+                throw new unoidl.com.sun.star.uno.Exception(ex.Message, ex);
+            }
+        }
```

Comment:
Take a look at how will your method look if you get rid of needless namespaces, type names and catch&rethrow:
```csharp
        private void SetActiveSheet(string sheetName)
        {
            var xSheets = MxDocument.getSheets();
            var xModel = (XModel)MxDocument;
            var xController = xModel.getCurrentController();
            var sheetView = (XSpreadsheetView)xController;
            var xSpreadsheet = (XSpreadsheet)xSheets.getByName(sheetName).Value;
            sheetView.setActiveSheet(xSpreadsheet);
        }
```